lib/kubernetes-deploy/runner.rb

```diff
@@ -76,11 +78,14 @@ def run
       phase_heading("Parsing deploy content")
       resources = discover_resources
 
+      tprs = resources.reject! { |r| r.tpr? }
+
       phase_heading("Checking initial resource statuses")
       resources.each(&:sync)
 
       phase_heading("Predeploying priority resources")
       predeploy_priority_resources(resources)
+      update_tprs(tprs)
```

Comment:
This minimally needs to happen before the predeploy, which can contain rails migrations that require the db. IMO it would be better if we can keep it part of the sequence though. Otherwise, we're circumventing the initial state discovery on line 84 and also skipping (or needing to repeat the implementation of) resource polling. How about augmenting `deploy_resources` to be able to handle tprs correctly, e.g. by checking whether any in the set are `.tpr?` and adjusting the command accordingly?